recipes/openfst/all/conanfile.py

```diff
@@ -54,47 +53,26 @@ class OpenFstConan(ConanFile):
         "enable_pdt": False,
         "enable_special": False,
     }
-
-    @property
-    def _settings_build(self):
-        return getattr(self, "settings_build", self.settings)
+    implements = ["auto_shared_fpic"]
 
     def export_sources(self):
         export_conandata_patches(self)
 
-    def config_options(self):
-        if self.settings.os == "Windows":
-            del self.options.fPIC
-
-    def configure(self):
-        if self.options.shared:
-            self.options.rm_safe("fPIC")
-
     def layout(self):
         basic_layout(self, src_folder="src")
 
+    def validate_build(self):
+        # Cross building is not supported. `configure` fails due to the existing AC_RUN_IFELSE autotools macro
+        # in the configure script which does not have a fallback in case of cross building
+        # There is neither a way of disabling the check nor a way of providing a fallback as `cross_compiling`
+        # variable is assigned on the fly in the configuration script
+        if hasattr(self, "settings_build") and cross_building(self):
```

Comment:
```suggestion
        if cross_building(self):
```
conan 2 always defines that attribute